hash_practice/exercises.py

```diff
@@ -2,18 +2,43 @@
 def grouped_anagrams(strings):
     """ This method will return an array of arrays.
         Each subarray will have strings which are anagrams of each other
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n)
+        Space Complexity: O(n)
     """
-    pass
+    dict = {}
+    
+    for word in strings:
+        sortedWord = "".join(sorted(word))
+        
+        if sortedWord in dict:
+            dict[sortedWord].append(word)
+            
+        else:
+            dict[sortedWord] = [word]
+            
+    return list(dict.values())
+        
 
 def top_k_frequent_elements(nums, k):
     """ This method will return the k most common elements
         In the case of a tie it will select the first occuring element.
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n)

```

Comment:
This is close, but you have to also account for `sorted` taking O(n*log(n)) time, which makes this overall O(n*log(n)).